# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/BaseTypes.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/compiler/translator/BaseTypes.h
--- a/gfx/angle/checkout/src/compiler/translator/BaseTypes.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/BaseTypes.h	Sat Sep 07 13:38:36 2019 +0000
@@ -622,6 +622,9 @@
 
     EvqDrawID,  // ANGLE_multi_draw
 
+    EvqBaseVertex,    // ANGLE_base_vertex_base_instance
+    EvqBaseInstance,  // ANGLE_base_vertex_base_instance
+
     // built-ins read by fragment shader
     EvqFragCoord,
     EvqFrontFacing,
@@ -690,6 +693,11 @@
     return (qualifier == EvqTemporary || qualifier == EvqGlobal);
 }
 
+inline bool IsStorageBuffer(TQualifier qualifier)
+{
+    return qualifier == EvqBuffer;
+}
+
 enum TLayoutImageInternalFormat
 {
     EiifUnspecified,
@@ -907,6 +915,8 @@
     case EvqPosition:               return "Position";
     case EvqPointSize:              return "PointSize";
     case EvqDrawID:                 return "DrawID";
+    case EvqBaseVertex:             return "BaseVertex";
+    case EvqBaseInstance:           return "BaseInstance";
     case EvqFragCoord:              return "FragCoord";
     case EvqFrontFacing:            return "FrontFacing";
     case EvqPointCoord:             return "PointCoord";